### PR TITLE
Restrict stream creation if the stream name has underscore

### LINKED ISSUE #4079: Restrict stream creation if the stream name has underscore
Given the Kubernetes validation for the app label and SCDF adds a label with the `stream name` prefix to the application, we can restrict the stream creation if the name has underscore in it.

Fixing this will avoid the stream deployment failure into K8s due to the regex validation failure on app label:
```
a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?'
```

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #4079

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java [hunk 1]
```diff
@@ -326,6 +331,14 @@ private void validateStreamDeploymentRequest(StreamDeploymentRequest streamDeplo
 				.findById(streamName)
 				.orElseThrow(() -> new NoSuchStreamDefinitionException(streamDeploymentRequest.getStreamName()));
 
+		if ("kubernetes".equals(platformInfo.getPlatformType())) {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java:334]: I understand that the platform info is **only** available at the time of stream deployment and hence we can't get this fixed at the stream definition controller. Maybe let's discuss if we can restrict the stream name with the below regex for all the platform types - that way this validation could move up at the stream creation itself (via Stream Definition Controller)

Comment (mminella) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java:334]: What's wrong with just making this a global rule and applying it to all platforms?

Comment (ilayaperumalg) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java:334]: I don't see any issue with that except some CF users might prefer the stream names with underscore in it ;)

Comment (mminella) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java:334]: For consistency and to be able to support the UI (where this validation is really needed), I'd think we want to just apply this rule globally.

**Comment by chrisjs** [comment]
@ilayaperumalg updated per discussion. also had to extend to tasks as the name gets inserted into labels the same way

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Pull request merged**